src/site/xdoc/userguide.xml

```diff
@@ -0,0 +1,181 @@
+        secureRandom.close();<br/>
+        </td>
+        </tr>
+        </table>
+
+
+        <h4>Usage of Cipher API</h4>
+        <p>Cipher provides an cryptographic interface for encryption and decryption. We provide two kind of implementations: JCE Cipher and Openssl Cipher. The JCE implementation uses JCE provider and the Openssl implementation uses Intel� AES New Instructions (Intel� AES NI). </p>
+        <h5>Usage of Byte Array Encryption/Decryption</h5>
+        <table>
+        <tr>
+        <td>
+        Properties properties = new Properties();<br/>
+        //Creates a Cipher instance with the transformation and properties.<br/>
+        Cipher cipher = Utils.getCipherInstance(CipherTransformation.AES_CTR_NOPADDING, properties);<br/><br/>
+        String input = "hello world!";<br/>
+        int inputOffset = 0;<br/>
+        int inputLen = input.length();<br/>
+        byte[] output = new byte[1024];<br/>
+        int outputOffset = 0;<br/>
+        //Initializes the cipher with ENCRYPT_MODE,key and iv.<br/>
+        cipher.init(Cipher.ENCRYPT_MODE, key, iv);<br/>
+        //Continues a multiple-part encryption/decryption operation for byte array.<br/>
+        cipher.update(input.getBytes("UTF-8"), inputOffset, inputLen, output, outputOffset);<br/>
+        //We should call do final at the end of encryption/decryption.<br/>
+        cipher.doFinal(inBuffer, outBuffer);<br/>
+        //Closes the cipher.<br/>
+        cipher.close();
+        </td>
+        </tr>
+        </table>
+
+        <h5>Usage of ByteBuffer Encryption/Decryption</h5>
+        <table>
+        <tr>
+        <td>
+        Properties properties = new Properties();<br/>
+        //Creates a Cipher instance with the transformation and properties.<br/>
+        Cipher cipher = Utils.getCipherInstance(CipherTransformation.AES_CTR_NOPADDING, properties);<br/><br/>
+        int bufferSize = 4096;<br/>
+        ByteBuffer inBuffer = ByteBuffer.allocateDirect(bufferSize);<br/>
+        ByteBuffer outBuffer = ByteBuffer.allocateDirect(bufferSize);<br/>
+        inBuffer.put("The data you want to encrypt or decrypt".getBytes("UTF-8"));<br/>
+        //Initializes the cipher with ENCRYPT_MODE,key and iv.<br/>
+        cipher..init(Cipher.ENCRYPT_MODE, key, iv);<br/>
+        //Continues a multiple-part encryption/decryption operation for byte array.<br/>
+        cipher.update(inBuffer, outBuffer);<br/>
+        //We should call do final at the end of encryption/decryption.<br/>
+        cipher.doFinal(inBuffer, outBuffer);<br/>
+        //Closes the cipher.<br/>
+        cipher.close();
+        </td>
+        </tr>
+        </table>
+
+
+        <h4>Usage of Stream API</h4>
+        <p>Stream provides the encryption and decryption data in stream manner. We provide CrypoInputStream,CTRCryptoInputStream and PositionedCryptoInputstream implementations for InputStream and CrypoOutputStream and CTRCryptoOutputStream implementations for OutputStream. </p>
```

Comment:
We provide CrypoInputStream,` `CTRCryptoInputStream, PositionedCryptoInputstream implementations for InputStream and CrypoOutputStream, CTRCryptoOutputStream implementations for OutputStream.